api/client/container/kill.go

```diff
@@ -0,0 +1,52 @@
+package container
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/api/client"
+	"github.com/docker/docker/cli"
+	"github.com/spf13/cobra"
+)
+
+type killOptions struct {
+	signal string
+
+	containers []string
+}
+
+// NewKillCommand creats a new cobra.Command for `docker kill`
+func NewKillCommand(dockerCli *client.DockerCli) *cobra.Command {
+	var opts killOptions
+
+	cmd := &cobra.Command{
+		Use:   "kill [OPTIONS] CONTAINER [CONTAINER...]",
+		Short: "Kill one or more running container",
+		Args:  cli.RequiresMinArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.containers = args
+			return runKill(dockerCli, &opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&opts.signal, "signal", "s", "KILL", "Signal to send to the container")
+	return cmd
+}
+
+func runKill(dockerCli *client.DockerCli, opts *killOptions) error {
+	var errs []string
+	for _, name := range opts.containers {
+		if err := dockerCli.Client().ContainerKill(context.Background(), name, opts.signal); err != nil {
```

Comment:
Ah, I forgot this one when updating context use…
@WeiZhang555 could you update this to share the context ? something like 

```go
ctx := context.Background()
for _, name := range opts.containers {
    if err := dockerCli.Client().ContainerKill(ctx, …) // …
}
```